modules/srtp/srtp.c

```diff
@@ -458,6 +492,49 @@ static int media_alloc(struct menc_media **stp, struct menc_sess *sess,
 }
 
 
+static int cmd_tx_rekey(struct re_printf *pf, void *arg) {
+	const struct cmd_arg *carg = arg;
+	struct call *selcall = NULL;
+	int err = 0;
+
+	if (!str_isset(carg->prm)) {
+		re_hprintf(pf, "usage: /srtprekey <call id>");
+		return EINVAL;
+	}
+
+	selcall = uag_call_find(carg->prm);
+	if (!selcall) {
+		re_hprintf(pf, "srtprekey: call id \"%s\" not found\n",
+			carg->prm);
+		return EINVAL;
+	}
+
+	re_hprintf(pf, "srtprekey: rekey srtp transmission keys of"
+		" call \"%s\"\n", carg->prm);
+
+	struct stream *astrm = audio_strm(call_audio(selcall));
+	struct stream *vstrm = video_strm(call_video(selcall));
+
+	if (astrm)
+		stream_remove_menc_media(astrm);
+	if (vstrm)
+		stream_remove_menc_media(vstrm);
+
+	err = call_update_media(selcall);
+	err |= call_modify(selcall);
+	if (err) {
+		re_hprintf(pf, "srtprekey: call update failed %m\n", err);
+	}
+
+	return err;
+}
+
+
+static const struct cmd cmdv[] = {
+{"srtprekey", 0, CMD_PRM, "Change outgoing streaming keys", cmd_tx_rekey},
```

Comment:
Is this command useful for endusers ?

Reply:
I need the transmission re-keying for the testcase anyway. I had in mind why not and its easy callable in the testcase.
To be fair its unlikely a user will ever re-key the outgoing stream with this command.